### PR TITLE
feat: add flag for using generator v2

### PR DESCRIPTION
**Pull request by Souvikns**
**Description**
In this PR I am adding a new Flag to use generator v2+, I am trying to change as little as possible from users original version of CLI. Currently most temapltes are not supporting v2 of generator, so it is better to have flag `--use-new-generator`, once more templates start supporting v2 of generator we can update the flag. 

```bash
asyncapi generate fromTemaplate ./spec.yaml @asyncapi/html-template --use-new-generator
```

**Related issue(s)**

See also #1388

### CODE REVIEW

File: check/asyncapi.md
Comment (Shurtu-gal) [inline check/asyncapi.md:1]: Where are these being used?

**Review by Shurtu-gal — APPROVED** [review]

**Comment by Shurtu-gal** [comment]
@Souvikns Please merge if ready.

**Comment by aeworxet** [comment]
@asyncapi/bounty_team

**Review by peter-rr — APPROVED** [review]
LGTM 👍

**Pull request merged**